trunk/src/form/form-edit.js

```diff
@@ -0,0 +1,123 @@
+import {__} from '@wordpress/i18n';
+import {Component} from '@wordpress/element';
+import {InspectorControls} from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
+
+
+import '../editor.scss';
+
+class FormEdit extends Component {
+
+    constructor() {
+        super(...arguments);
+        this.state = {
+            dropdownItems: [],
+            showDropdown: false,
+            interactive: false,
+        };
+
+        // This binding is necessary to make `this` work in the functions
+        this.render = this.render.bind(this);
+        this.hideOverlay = this.hideOverlay.bind(this);
+    }
+
+    componentDidMount() {
+
+        // We are using the pre-existing PHP Ajax function to allow us to obtain the user's forms.
+        // We need to parse the return value, as the response comes backs as HTML (<select><option>...).
+        let xhr = new XMLHttpRequest()
+
+        xhr.addEventListener('load', () => {
+            let shortcodes = [...xhr.responseText.matchAll(/"\[.*?]"/g)]
+            let dropdownItems = [];
+            for (let i = 0; i < shortcodes.length; i++) {
+                let parsedShortcode = [...shortcodes[i][0].matchAll(/ (.*?)=&quot;(.*?)&quot;/g)];
+                dropdownItems.push({form_id: parsedShortcode[0][2], title: parsedShortcode[1][2]})
+            }
+
+            if (shortcodes.length > 0) {
+                this.setState({
+                    dropdownItems: dropdownItems,
+                    showDropdown: true,
+                });
+            }
+
+        })
+
+        xhr.open('POST', ajaxurl)
+        let formData = new FormData();
+        formData.append("action", "get_pardot_forms_shortcode_select_html");
+        xhr.send(formData);
+    }
+
+    // We need getDerivedStateFromProps() and hideOverlay() to allow the user to click anywhere on the block to select it.
+    // I believe the iframe causes issues with the way the Block Editor interacts with user clicks.
+    // We essentially need to create an overlay to capture the first click, then switch to interactive mode,
+    // which allows the user to interact with the iframe.
+    // Source: https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/embed/embed-preview.js#L34
+    static getDerivedStateFromProps(nextProps, state) {
+        if (!nextProps.isSelected && state.interactive) {
+            return {interactive: false};
+        }
+
+        return null;
+    }
+
+    hideOverlay() {
+        this.setState({interactive: true});
+    }
+
+    handleValueOnChange(e) {
+        this.props.setAttributes({
+            [e.target.id]: e.target.value
```

Comment:
interesting that we have to set this as an array of the ID -- is that expected?

Reply:
if this is a string anyway, can't we just use string interpolation to set the key?